channel/handlers_client.go

```diff
@@ -1334,28 +1334,175 @@ func (server *Server) playerUseCash(conn mnet.Client, reader mpacket.Reader) {
 	slot := reader.ReadInt16()
 	itemID := reader.ReadInt32()
 
+	used := false
+
 	switch itemID {
 	case constant.ItemSafetyCharm:
+		// Safety charm prevents exp loss on death - stored in player state
+		// Implementation would require death handling modification
+		log.Printf("Safety Charm not fully implemented")
```

Comment:
This can be implemented by just add a safety charm flag to player struct and if they die then dont remove exp and then set the flag back to false